.circleci/config.yml

```diff
@@ -528,132 +602,141 @@ jobs:
     executor: default-executor
     steps:
       - git-checkout
-      - provision
+      - restore-workspace
```

Comment:
It turns out restoring workspace is faster than just restoring a typical cache. This is what lets us use slimmer docker images for testing and ultimately led the lowest start up cost. 

It takes about 20s to setup using restore workspace, and it would take about 1 minute to restore via a base image with minimal install overhead, and it would take about 2 to 3 minutes to install and via yarn cache restore and a yarn install. Basically, this restore-workspace seem like the optimal approach especially if we can have a build stage that gates other operations.